src/cbs.c

```diff
@@ -167,57 +356,102 @@ int cbs_close(CBS_HANDLE cbs)
 
     if (cbs == NULL)
     {
+        /* Codes_SRS_CBS_01_045: [ If the argument `cbs` is NULL, `cbs_close` shall fail and return a non-zero value. ]*/
+        LogError("NULL cbs handle");
+        result = __FAILURE__;
+    }
+    else if (cbs->cbs_state == CBS_STATE_CLOSED)
+    {
+        /* Codes_SRS_CBS_01_047: [ `cbs_close` when closed shall fail and return a non-zero value. ]*/
+        /* Codes_SRS_CBS_01_048: [ `cbs_close` when not opened shall fail and return a non-zero value. ]*/
+        LogError("Already closed");
         result = __FAILURE__;
     }
     else
     {
+        if (cbs->cbs_state == CBS_STATE_OPENING)
+        {
+            /* Codes_SRS_CBS_01_079: [ If `cbs_close` is called while OPENING, the `on_cbs_open_complete` callback shall be called with `CBS_OPEN_CANCELLED`, while passing the `on_cbs_open_complete_context` as argument. ]*/
+            cbs->on_cbs_open_complete(cbs->on_cbs_open_complete_context, CBS_OPEN_CANCELLED);
```

Comment:
Maybe put this after the close below.

---
In reply to: [107039232](https://github.com/Azure/azure-uamqp-c/pull/56#discussion_r107039232) [](ancestors = 107039232)